### PR TITLE
Feature/facets

### PR DESCRIPTION
**Pull request by vitalie-ucommerce**
This PR is mainly just for discarding `SearchLibrary`
I will activate it, after @skovsboll PR is merged.

Relates: https://github.com/Ucommercenet/Ucommerce/pull/226

### CODE REVIEW

File: src/Ucommerce.Avenue.Umbraco/Search/AvenueProductIndexDefinition.cs [hunk 1]
```diff
@@ -12,20 +11,10 @@ public AvenueProductIndexDefinition() : base()
             this.Field(p => p["CollarSize"], typeof(string));
             this.Field(p => p["ShoeSize"], typeof(string));
             this.Field(p => p["Colour"], typeof(string));
-
+            this.PricesField(p => p.UnitPrices.ToString());
```
Comment (skovsboll) [inline src/Ucommerce.Avenue.Umbraco/Search/AvenueProductIndexDefinition.cs]: Hmmm....
We don't want to limit the number of price dictionaries that a customer could add. Already, we talked about adding YourPrice in addition to ListPrice to the default product index. It's very likely to happen.
Can there be more than one PricesField?

Comment (skovsboll) [reply src/Ucommerce.Avenue.Umbraco/Search/AvenueProductIndexDefinition.cs]: Also, what's the `.ToString()` doing there. Looks a bit odd.

File: src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs [hunk 2]
```diff
@@ -104,7 +103,7 @@ private IList<ProductViewModel> MapProductsInCategories(Category category)
 
             var subCategories = CatalogLibrary.GetCategories(category.Categories);
             var products =
-                SearchLibrary.GetProductsFor(category.Categories.Append(category.Guid).ToList(), facetsForQuerying);
+                CatalogLibrary.GetProductsFor(category.Categories.Append(category.Guid).ToList(), facetsForQuerying.ToFacetDictionary());
```
Comment (skovsboll) [inline src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs]: Why does it require `.ToList()`? Could we make the API more friendly by changing parameter type to `IEnumerable<Guid>`?

File: src/Ucommerce.Avenue.Umbraco/Controllers/FacetsController.cs [hunk 3]
```diff
@@ -62,7 +60,7 @@ public ActionResult Index()
 
             if (ShouldDisplayFacets(category))
             {
-                IList<Facet> facets = _searchLibrary.GetFacetsFor(category.Guid, facetsForQuerying).Facets;
+                IList<Facet> facets = CatalogLibrary.GetProducts(category.Guid, facetsForQuerying.ToFacetDictionary()).Facets;
```
Comment (skovsboll) [inline src/Ucommerce.Avenue.Umbraco/Controllers/FacetsController.cs]: Did you find a way to return just the facets, so we don't have to throw good data (the products) in the bin?

Comment (vitalie-ucommerce) [reply src/Ucommerce.Avenue.Umbraco/Controllers/FacetsController.cs]: good point, I added it

File: src/Ucommerce.Avenue.Umbraco/Search/AvenueProductIndexDefinition.cs (hunk 1 above)
Comment (vitalie-ucommerce) [reply src/Ucommerce.Avenue.Umbraco/Search/AvenueProductIndexDefinition.cs]: you are right, I forgot about this

File: src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs (hunk 2 above)
Comment (vitalie-ucommerce) [reply src/Ucommerce.Avenue.Umbraco/Controllers/CategoryController.cs]: It's actually harder than just changing the type.
We need to add a way to handle `IEnumrable` in our API.
So if we really want this, it's going to be another PR

**Pull request merged**